lib/logstash/inputs/tcp.rb

```diff
@@ -146,6 +151,7 @@ def server_connection_thread(output_queue, socket)
         @logger.debug? && @logger.debug("Accepted connection", :client => s.peer, :server => "#{@host}:#{@port}")
         handle_socket(s, s.peeraddr[3], s.peeraddr[1], q, @codec.clone)
       ensure
+        metric.decrement(:connections)
```

Comment:
I am wondering if it would make testing easier if we move the metric decrement in the `#delete_connection_socket` method.